graphql/resolvers/exerciseCrud.ts

```diff
@@ -91,3 +92,67 @@ export const deleteExercise = async (
     }
   })
 }
+
+export const flagExercise = async (
+  _parent: void,
+  arg: MutationFlagExerciseArgs,
+  { req }: Context
+): Promise<Exercise> => {
+  const { id } = arg
+
+  const authorId = req.user?.id
+  if (!authorId) throw new Error('No User')
```

Comment:
Protip for doing this:

1. Create a helper function, we usually call them `containers`

```js
export const withUserContainer = (cb) => {
  return async (
     _parent: void,
     arg: ...,
     { req }: Context
   ): Promise... => {
         const { id } = arg
         const userId = req.user?.id
         if (!authorId) throw new Error('No User')
         return cb(_parent, arg, {req, userId} )
}
```

Then now you can use this container for every resolver you would wish to make sure userId exists:

```js

export const flagExercise = withUserContainer(async () => {} )

export const removeExerciseFlag = withUserContainer( async () => {} )
```